### PR TITLE
Switch to admin API in tctl update namespace

### PR DESCRIPTION
**Pull request by feedmeapples**
**What changed?**

Switches to Admin API when updating namespaces through `tctl admin namespace update`

**Why?**

Simplifies tctl usage by removing direct DB connection params

**How did you test it?**

```
$ tctl --ns 1 admin namespace update --description 222
Namespace 1 successfully updated
$ ./tctl admin namespace list
... Name:1,Description:222222222222...
```

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: service/frontend/adminHandler.go [hunk 1]
```diff
@@ -264,7 +264,7 @@ func (adh *AdminHandler) UpdateNamespace(ctx context.Context, request *adminserv
 	}
 
 	if request.GetNamespace() == "" {
-		return nil, adh.error(errNamespaceNotSet, scope)
+		return nil, adh.error(interceptor.ErrNamespaceNotSet, scope)
 	}
```
Comment (wxing1292) [inline service/frontend/adminHandler.go:268]: the new interceptor should handle this case?

File: tools/cli/adminCommands.go [hunk 2]
```diff
@@ -209,6 +213,141 @@ func AdminListNamespaces(c *cli.Context) {
...
+		activeCluster := c.String(FlagActiveClusterName)
+		fmt.Printf("Will set active cluster name to: %s, other flag will be omitted.\n", activeCluster)
+		replicationConfig := &replicationpb.NamespaceReplicationConfig{
+			ActiveClusterName: activeCluster,
+		}
+		req = &adminservice.UpdateNamespaceRequest{
+			Namespace:         namespace,
+			ReplicationConfig: replicationConfig,
+		}
+	} else {
```
Comment (wxing1292) [inline tools/cli/adminCommands.go:236]: preferable to let server make the decision and / or return error

File: tools/cli/adminCommands.go [hunk 3]
```diff
@@ -209,6 +213,141 @@ func AdminListNamespaces(c *cli.Context) {
...
+		resp, err := client.DescribeNamespace(ctx, &workflowservice.DescribeNamespaceRequest{
+			Namespace: namespace,
+		})
+		if err != nil {
+			if _, ok := err.(*serviceerror.NotFound); !ok {
+				return fmt.Errorf("namespace update failed: %s", err)
+			} else {
+				return fmt.Errorf("namespace %s does not exist: %s", namespace, err)
+			}
+		}
```
Comment (wxing1292) [inline tools/cli/adminCommands.go:246]: 1. this logic should be handled by server?

the API is called UpdateNamespace, meaning caller should only send out variables to be updated. (send out diff)
if the API is called SetNamespace/PutNamespace, then caller is responsible for telling the sever exactly how a namespace should looks like. (send out entire namespace)

there are some limitation of the update API, e.g. hard to specify `delete one entry from a slice map`

so what is the intention? update or set / put?

**Comment by wxing1292** [comment]
you may also want to sync with @alexshtin, he is doing the namespace deletion

**Comment by feedmeapples** [comment]
as per discussion regarding Operator APIs vs Application API, + given that the namespace APIs are now all part of workflowservice, closing this PR and going to remove namespace APIs from adminservice

@yiminc @alexshtin @wxing1292

**Pull request closed without merging**